packages/amplify-provider-awscloudformation/src/aws-utils/aws-lambda.ts

```diff
@@ -41,15 +40,24 @@ export class Lambda {
     return result;
   }
 
-  public async deleteLayerVersions(layerNameOrArn: string, versions: number[]) {
+  async deleteLayerVersions(layerNameOrArn: string, versions: number[]) {
     const params = { LayerName: layerNameOrArn, VersionNumber: undefined };
-    for (let version of versions) {
+    const deletionPromises = [];
+    for (const version of versions) {
       params.VersionNumber = version;
-      try {
-        await this.lambda.deleteLayerVersion(params).promise();
-      } catch (e) {
-        this.context.print.error(e); // TODO full error handling
-      }
+      deletionPromises.push(this.lambda.deleteLayerVersion(params).promise());
+    }
+
+    try {
+      await Promise.all(deletionPromises);
+    } catch (e) {
+      this.context.print.error(
+        'Failed to delete some or all layer versions. Check your internet connection and try again. ' +
```

Comment:
If a customer ends up in this state, how difficult is the recovery path? Could they really "just try again" and if it works they're good?

Reply:
I have never actually run into a state where this happens, so I'm not certain. But the code will pull the latest layer information from the cloud, so I think that is indeed the case.